inference/core/workflows/execution_engine/dynamic_blocs/entities.py

```diff
@@ -0,0 +1,130 @@
+from enum import Enum
+from typing import Any, Dict, List, Literal
+
+from pydantic import BaseModel, Field
+
+
+class SelectorType(Enum):
+    INPUT_IMAGE = "input_image"
+    INPUT_PARAMETER = "input_parameter"
+    STEP_OUTPUT = "step_output"
+
+
+class ValueType(Enum):
+    ANY = "any"
+    INTEGER = "integer"
+    FLOAT = "float"
+    BOOLEAN = "boolean"
+    DICT = "dict"
+    LIST = "list"
+    STRING = "string"
+
+
+class DynamicInputDefinition(BaseModel):
+    type: Literal["DynamicInputDefinition"]
+    has_default_value: bool = Field(
+        default=False,
+        description="Flag to decide if default value is provided for input",
+    )
+    default_value: Any = Field(
+        description="Definition of default value for a field. Use in combination with, "
+        "`has_default_value` to decide on default value if field is optional.",
+        default=None,
+    )
+    is_optional: bool = Field(
+        description="Flag deciding if `default_value` will be added for manifest field annotation.",
+        default=False,
+    )
+    is_dimensionality_reference: bool = Field(
+        default=False,
+        description="Flag deciding if declared property holds dimensionality reference - see how "
+        "dimensionality works for statically defined blocks to discover meaning of the "
+        "parameter.",
+    )
+    dimensionality_offset: int = Field(
+        default=0,
+        ge=-1,
+        le=1,
+        description="Accepted dimensionality offset for parameter. Dimensionality works the same as for "
+        "traditional workflows blocks.",
+    )
+    selector_types: List[SelectorType] = Field(
```

Comment:
How do you accept an image (whether it be an output of a previous step or an input to the workflow)?

Reply:
u choose union of 
```
class SelectorType(Enum):
    INPUT_IMAGE = "input_image"
    INPUT_PARAMETER = "input_parameter"
    STEP_OUTPUT = "step_output"
```
which is almost 1-1 mapping to types: `WorkflowImageSelector`, `WorkflowParameterSelector`, `StepOutputSelector`